backend-rust/src/graphql_api/account.rs

```diff
@@ -758,14 +763,23 @@ impl Account {
         Ok(account)
     }
 
-    pub async fn query_by_address(pool: &PgPool, address: String) -> ApiResult<Option<Self>> {
+    pub async fn query_by_address(
+        pool: &PgPool,
+        account_address: String,
+    ) -> ApiResult<Option<Self>> {
+        let Ok(account_address) = concordium_rust_sdk::base::contracts_common::AccountAddress::from_str(
+            &account_address,
+        ) else {
+            return Ok(None);
+        };
```

Comment:
Returning an error seems more informative, then they would know the address to have some typo.

Reply:
Also seems consistent with what we do.

Reply:
2025-04-23T12:53:02.737778Z  INFO request:execute:field{path=accountByAddress parent_type=Query return_type=Account!}: async_graphql::graphql: error error=Could not find resource

We are getting a graphql error, when there is absolutely no error when a user requests a none existing account.

That said as mentioned above it is consistent with current behaviour, so lets just do it.